03-transactions.md

```diff
@@ -315,28 +394,32 @@ Thus, a simplified formula for *expected weight* is used, which assumes:
 * Signatures are 73 bytes long (the maximum length).
 * There are a small number of outputs (thus 1 byte to count them).
 * There are always both a `to_local` output and a `to_remote` output.
+* (if `option_anchor_outputs`) there are always both a `to_local_anchor` and `to_remote_anchor` output.
 
 This yields the following *expected weights* (details of the computation in [Appendix A](#appendix-a-expected-weights)):
 
-    Commitment weight:   724 + 172 * num-untrimmed-htlc-outputs
-    HTLC-timeout weight: 663
-    HTLC-success weight: 703
+    Commitment weight (no option_anchor_outputs):   724 + 172 * num-untrimmed-htlc-outputs
+    Commitment weight (option_anchor_outputs):     1124 + 172 * num-untrimmed-htlc-outputs
+    HTLC-timeout weight (no option_anchor_outputs): 663
+    HTLC-timeout weight (option_anchor_outputs): 666
+    HTLC-success weight (no option_anchor_outputs): 703
+    HTLC-success weight (option_anchor_outputs): 706
 
 Note the reference to the "base fee" for a commitment transaction in the requirements below, which is what the funder pays. The actual fee may be higher than the amount calculated here, due to rounding and trimmed outputs.
```

Comment:
@rustyrussell, your suggestion is: (making it an "anchored" github comment, so it can be resolved and the PR doesn't become even bigger)

> +Note the reference to the "base fee" for a commitment transaction in the requirements below, which is what the funder pays. The actual fee may be higher than the amount calculated here, due to rounding and trimmed outputs, or may be lower, as the base fee also pays for the anchor outputs (if `option_anchor_outputs`).

I don't understand how it can be lower? The base fee takes into account two anchors, but there may be only one, in which case the fee rate will turn out to be higher?

Reply:
Because this redefines "base fee" as "tx fee + anchor outputs fee".  Actual fee may be larger than "tx fee", but still less than "tx fee + anchor outputs fee".

Which is confusing, so let's go back to your original, where the anchor fee is added during construction, perhaps with this change to make it a bit clearer?

```diff
-2. Calculate the base [commitment transaction fee](#fee-calculation).
+2. Set fee to the base [commitment transaction fee](#fee-calculation).
-3. Subtract this base fee from the funder (either `to_local` or `to_remote`).
-If `option_anchor_outputs` applies to the commitment transaction,
-also subtract two times the fixed anchor size of 330 sats from the funder
-(either `to_local` or `to_remote`).
+3. If `option_anchor_outputs` applies to the commitment transaction:
+   - Add 660 satoshis to the fee.
+4. Subtract this fee from the funder (either `to_local` or `to_remote`).
```

(Yeah, the rest need to be renumbered now, sorry :()

Reply:
I think it could be confusing to make the anchors part of the fee. Most readers probably think of 'miner fee' if they see 'fee'?